src/test/regress/expected/isolation_copy_vs_all_on_mx.out

```diff
@@ -290,3 +290,78 @@ stop_session_level_connection_to_node
 
 (1 row)
 
+
+starting permutation: s1-start-session-level-connection s1-begin-on-worker s1-copy s2-start-session-level-connection s2-worker-create-index-concurrently s1-commit-worker s2-stop-connection s3-select-count s1-stop-connection
+step s1-start-session-level-connection:
+        SELECT start_session_level_connection_to_node('localhost', 57637);
+
+start_session_level_connection_to_node
+---------------------------------------------------------------------
+
+(1 row)
+
+step s1-begin-on-worker:
+        SELECT run_commands_on_session_level_connection_to_node('BEGIN');
+
+run_commands_on_session_level_connection_to_node
+---------------------------------------------------------------------
+
+(1 row)
+
+step s1-copy:
+ SELECT run_commands_on_session_level_connection_to_node('COPY copy_table FROM PROGRAM ''echo 5, 50 && echo 6, 60 && echo 7, 70''WITH CSV');
+
+run_commands_on_session_level_connection_to_node
+---------------------------------------------------------------------
+
+(1 row)
+
+step s2-start-session-level-connection:
+        SELECT start_session_level_connection_to_node('localhost', 57638);
+
+start_session_level_connection_to_node
+---------------------------------------------------------------------
+
+(1 row)
+
+step s2-worker-create-index-concurrently:
+ SELECT run_commands_on_session_level_connection_to_node('CREATE INDEX CONCURRENTLY copy_table_index ON copy_table(id)');
```

Comment:
I'm surprised this works, since I would expect the CREATE INDEX CONCURRENTLY to have to wait on the transaction that runs the run_commands_on_session_level_connection_to_node call. Do we have the coordinator in the metadata here?

Reply:
we should probably disallow index commands from the worker node, when coordinator is not in the metadata, if we don't do so already

Reply:
We have `citus.allow_unsafe_locks_from_workers` that is used  in `AcquireDistributedLockOnRelations`, which I guess should cover what you have in mind? The default is false, which seems to fit what you want.

Reply:
CREATE INDEX CONCURRENTLY mainly waits for snapshots, not so much locks

Hence, I'm surprised this test works, because the transaction doing SELECT run_commands_on_session_level_connection_to_node is likely to have a snapshot. 

Hence, I'm wondering whether the reason this works is that the CREATE INDEX CONCURRENTLY commands is not getting propagated to the coordinator.

Hence, my question: Do we have the coordinator in the metadata here? 

We should anyway disallow DDL from worker when we don't.

Reply:
> We should anyway disallow DDL from worker when we don't.

Yes, agreed. We currently throw the following error for `CREATE INDEX`

```SQL
create index on t1 (a);
ERROR:  Cannot acquire a distributed lock from a worker node since the coordinator is not in the metadata.
HINT:  Either run this command on the coordinator or add the coordinator in the metadata by using: SELECT citus_set_coordinator_host('<hostname>', <port>);
Alternatively, though it is not recommended, you can allow this command by running: SET citus.allow_unsafe_locks_from_workers TO 'on';
```

> Hence, my question: Do we have the coordinator in the metadata here?

Probably not, we have seem to have a bug, we allow `CREATE INDEX CONCURRENTLY` even without coordinator in the metadata

```SQL
-- same cluster, query form worker
create index concurrently on t1 (a);
CREATE INDEX
```

Reply:
> Hence, I'm wondering whether the reason this works is that the CREATE INDEX CONCURRENTLY commands is not getting propagated to the coordinator.

And, the answer is yes for the current patch, which is  a bug

Reply:
> Let me add the logic to disallow ddl from any node when coordinator is not in metadata. 

I think we should consider sing `citus.allow_unsafe_locks_from_workers`, which is already the case for CREATE INDEX etc.